src/app/services/cohort.service.ts

```diff
@@ -189,34 +164,48 @@ export class CohortService {
   }
 
   public updateCountsWithAllCohorts(): Promise<any> {
-    console.log('Updating counts from all cohorts...');
-    this.isUpdatingAll = true;
-    let combination: CombinationConstraint = new CombinationConstraint();
-    combination.combinationState = CombinationState.And;
-    combination.mark = ConstraintMark.SUBJECT;
-    this.cohorts.forEach((cohort: Cohort) => {
-      if (cohort.selected) {
-        combination.addChild(cohort.constraint);
+    return new Promise((resolve, reject) => {
+      console.log('Updating counts from all cohorts...');
+      this.isUpdatingAll = true;
+      let combination: CombinationConstraint = new CombinationConstraint();
+      combination.combinationState = CombinationState.And;
+      combination.mark = ConstraintMark.SUBJECT;
+      this.cohorts.forEach((cohort: Cohort) => {
+        if (cohort.selected) {
+          combination.addChild(cohort.constraint);
+        }
+      });
+      if (this.saveSubjectSetBeforeUpdatingCounts) {
+        this.resourceService.saveSubjectSet('temp', combination).subscribe((subjectSet: SubjectSet) => {
```

Comment:
if the 'combination' constraint has been used before, there would be no need to save another subjectSet, right? since a subjectSet has been saved previously for that constraint

Reply:
probably involves a more extensive discussion